### DISCUSSION TITLE
Question about GSEA. How to rank?

### DISCUSSION THREAD

**Discussion by Sirin24** [satijalab/seurat#9291] [General]
Hello, how to rank genes for GSEA? I have disease fibroblasts and control fibroblasts and I ran findmarkers() on them. 

is the avg_log2fc sufficient to rank the genes for GSEA using clusterprofiler package? On bioconductor support page they advice against filtering out genes out with high pvalue, meaning we should keep all genes, so how to rank them for gene set enrichment and not over representation analysis?

**Comment by shijianlvxing-lab** [comment]
Yes, all genes should be preserved, but they should be sorted from the largest to the smallest based on avg_log2fc.

**Comment by Sirin24** [comment]
I am not sure as Seurat findmarkers() has had some changes so I would appreciate it if someone from satijalab could help me.

**Comment by samuel-marsh** [comment]
So you aren't going to want to just use avg_log2fc in the absence of significance measure because genes with log fold changes above a certain cutoff but not significant is not what you want to be testing.

Also presuming that you ran cell-level DE with FIndMarkers you also are not going to want to use those pvalues because they are not statistically valid due to double dipping.

In the meantime though I'm going to move this issue to Discussions as it is not about the Seurat package but about analysis using a different package.  Hopefully more from community can then weigh in for you.

Best,
Sam

**Comment by Sirin24** [comment]
thanks @samuel-marsh so this means that I should not be using Seurat if I will be doing Gene set enrichment analysis?
